src/main/java/net/neoforged/srgutils/InternalUtils.java

```diff
@@ -30,32 +28,41 @@ static IMappingFile load(InputStream in) throws IOException {
     }
 
     static INamedMappingFile loadNamed(InputStream in) throws IOException {
-        List<String> lines = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8)).lines()
-            //.map(InternalUtils::stripComment)
-            .filter(l -> !l.isEmpty()) //Remove Empty lines
-            .collect(Collectors.toList());
-
-        if (lines.isEmpty())
-            return IMappingBuilder.create().build();
-
-        String firstLine = lines.get(0);
-        Iterator<String> itr = lines.iterator();
-        while (stripComment(firstLine).isEmpty() && itr.hasNext())
-            firstLine = itr.next();
+        LineNumberReader reader = new LineNumberReader(new InputStreamReader(in));
+        reader.mark(2048);
```

Comment:
I've discussed this PR interally for a bit, to ask for second opinions, and a concern has been raised with the `mark(2048)` and `reset()` pair being potentially problematic if the first 2048 bytes of the mapping file do not contain substantive mapping data (which may happen, for example, with a large comment).

Is that an acceptable risk, or should we increase the mark size to something quite large, to prevent that possibility?

Reply:
We can avoid this by marking before each header line we test for comments. The only difference then would be that loadSlimSRG will not receive empty/comment header lines.